lib/ruby_lsp/server.rb

```diff
@@ -196,6 +206,21 @@ def load_addons(include_project_addons: true)
 
     private
 
+    #: (Hash[Symbol, untyped] message) -> void
+    def workspace_configuration_did_change(message)
```

Comment:
My understanding based on the [spec](https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#didChangeConfigurationParams) is that the parameters of `workspace/didChangeConfiguration` already include which settings are supposed to be changed.

Why do we need to make a request to the client? Could we use what we receive as parameters instead?

Reply:
The way I understand the spec is that `workspace/didChangeConfiguration` can include the payload, but it's soft-deprecated behaviour, and I chose not to implement it.

> The workspace/configuration request is sent from the server to the client to fetch configuration settings from the client.  [...] This pull model replaces the old push model were the client signaled configuration change via an event.

https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#workspace_configuration

I think it's a strategic decision for the maintaining team to choose which approach to support and I will be happy to adjust this PR to whatever you decide (push, pull or both).

Reply:
Got it. That makes sense, but then let's only implement the pull model rather than pulling inside of the old event.

We'll probably need to fire this request upon receiving the `initialize` notification, so that we can read any other configurations.

Unfortunately, it's a bit wasteful for editors that include workspace settings as part of the initial boot. I wonder if there's any information we can use to send this request selectively.

Reply:
The reason why I implemented it in response to `didChangeConfiguration` is that Node's LS testbed [seems to be doing that](https://github.com/microsoft/vscode-languageserver-node/blob/cd32d3e505e8ec8df660bc8e328d5f650673c2d8/testbed/server/src/server.ts#L240-L247) - however I don't see it in actually real server implementation, so I'm not sure what's going on with this.

You are right about this being unnecessary for majority of LSP connections that do not want to dynamically reconfigure anything. Maybe it's not worth implementing after all?